kt/godot-runtime/src/main/kotlin/godot/core/TypeManager.kt

```diff
@@ -1,19 +1,25 @@
 package godot.core
 
+import godot.util.VoidPtr
+
 object TypeManager {
-    private val userTypes = HashSet<String>()
+    val userTypeNames = LinkedHashSet<String>()
+    val userTypeConstructors = mutableListOf<(VoidPtr) -> KtObject>()
 
     val engineTypeNames = LinkedHashSet<String>()
-    val engineTypesConstructors = mutableListOf<() -> KtObject>()
+    val engineTypeConstructors = mutableListOf<() -> KtObject>()
```

Comment:
Encapsulation is broken here, this should be all private. Why do we need to expose a mutable field to the public?.